src/main/java/prefs/Preferences.java

```diff
@@ -128,15 +138,32 @@ public void setLastOpenBoard(String board) {
     }
     
     public Optional<String> switchBoard() {
-        if (getLastOpenBoard().isPresent() && getAllBoards().size() > 1) {
-            List<String> boardNames = getAllBoardNames();
+        setBoardToSwitch();
+        return getLastOpenBoard();
+    }
+
+    /**
+     * Sets the lastOpenBoard to the board to switch to
+     */
+    private void setBoardToSwitch(){
```

Comment:
Board switching should not even be in this class! :-o This class is for managing preferences!! Oh man. @garbanzos I know you are not the one who put it there, but can you refactor board switching out of here? @dariusf can you advise @garbanzos on the direction to take? It should be similar to label picker, but doesn't have a UI at the moment.